src/index.ts

```diff
@@ -146,40 +151,63 @@ function useDetectScroll(props: ScrollProps = {}): ScrollInfo {
 
   /** Function to update scroll direction */
   const updateScrollDir = useCallback(() => {
-    const scroll = axis === Axis.Y ? window.scrollY : window.scrollX;
+    const scroll =
+      target instanceof Window
+        ? axis === Axis.Y
+          ? window.scrollY
+          : window.scrollX
+        : axis === Axis.Y
+          ? target.scrollTop
+          : target.scrollLeft;
 
     if (Math.abs(scroll - lastScroll.current) >= threshold) {
       setScrollDir(scroll > lastScroll.current ? scrollDown : scrollUp);
       lastScroll.current = Math.max(0, scroll);
     }
     ticking.current = false;
-  }, [axis, threshold, scrollDown, scrollUp]);
+  }, [target, axis, threshold, scrollDown, scrollUp]);
 
   useEffect(() => {
     /** Function to update scroll position */
     const updateScrollPosition = () => {
-      const top = window.scrollY;
-      const left = window.scrollX;
+      const top = target instanceof Window ? target.scrollY : target.scrollTop;
+      const left =
+        target instanceof Window ? target.scrollX : target.scrollLeft;
       const bottom =
-        document.documentElement.scrollHeight - window.innerHeight - top;
+        target instanceof Window
+          ? document.documentElement.scrollHeight - window.innerHeight - top
+          : document.documentElement.scrollHeight - target.scrollHeight - top;
       const right =
-        document.documentElement.scrollWidth - window.innerWidth - left;
+        target instanceof Window
+          ? document.documentElement.scrollWidth - window.innerWidth - left
+          : document.documentElement.scrollHeight - target.scrollWidth - left;
 
       setScrollPosition({ top, bottom, left, right });
     };
 
     /** Call the update function when the component mounts */
     updateScrollPosition();
 
-    window.addEventListener('scroll', updateScrollPosition);
+    target instanceof Window
```

Comment:
You can prevent using ternary operations here as well by doing something like below:

```ts
    const targetElement = target as EventTarget;
    targetElement.addEventListener('scroll', updateScrollPosition);

    return () => {
      targetElement.removeEventListener('scroll', updateScrollPosition);
    };
```